### DISCUSSION TITLE
Automation processes on the ems-esp itself

### DISCUSSION THREAD

**Discussion by yazend** [emsesp/EMS-ESP32#1480] [Ideas]
It would be very helpful to be able to carry out some "if -> then --> else" functions on the ems-esp itself, especially when it comes to switching cycles of the heating, I don't like to leave this to an external system, I mention in the example. Homeassistant is further referred to as HA, but this applies to everything else as well.

Monitoring the room temperature -> the temperature falls below 3°C -> switch the heating from summer mode to winter mode -> until 6°C is reached -> then switch again from winter mode to summer mode.

If I implement this in HA, the automations must be mutually locked, and it must also be ensured that no backup is being created if you are in the range of 3.2°C to 3.0°C, otherwise the automation will not triggerd.
I would therefore like to outsource this from the home automation center, as heating is a priority and no susceptibility to errors should be tolerated here.

Therefore it would be very helpful if you could create this function in the scheduler.

Example:
If sensor "Sensor ID: 24-C203-81E4-E23C" is equal or less than 3°C
Then switch from “Summer mode to Winter mode”
and switch from "manual mode to auto mode"
Else (optional settings)

If sensor "Sensor ID: 24-C203-81E4-E23C" equals or greater than 6°C
Then switch from “Winter mode to Summer mode”
and switch from "Auto mode to manual mode"
Else (optional settings)

BR yazend

**Comment by proddy** [comment]
interesting. thinking. Any more votes?

**Comment by bbqkees** [comment]
I think this goes a bit outside of the intended scope of EMS-ESP (which is basically to have the decision making outside of EMS-ESP) but as Proddy and Michael introduced the scheduler in the mean time it's more of a grey area. If-then-else combined with the scheduler could be an interesting option for simple statements.
However, if you implement if-then-else in EMS-ESP, there are a lot of things to consider like how complicated or nested can it be, ESP32 chip resources etc. And what if both f.i. HA and EMS-ESP are 'if-then-elsing' the same boiler entity simultaneously this could get messy.
